lib/util/pthread_create_blocking.h

```diff
@@ -0,0 +1,20 @@
+#ifndef _PTHREAD_CREATE_BLOCKING_H_
+#define _PTHREAD_CREATE_BLOCKING_H_
+
+#include <pthread.h>
+
+/**
+ * pthread_create_blocking(thread, attr, init_routine, run_routine,
+ *     cleanup_routine, arg):
+ * Run pthread_create() and block until the new ${thread} has started and
+ * ${init_routine} has finished.  ${init_routine} and ${cleanup_routine} may
+ * be NULL.  If ${init_routine} is not NULL and returns a non-zero value, do
+ * not attempt to run ${run_routine} and instead return that value.  When
+ * ${run_routine} finishes, make its returned value available via
+ * pthread_join().
```

Comment:
This comment tells me nothing about what `cleanup_routine` does except that it may be NULL.